inference-engine/src/mkldnn_plugin/ngraph_transformations/align_matmul_input_ranks.cpp

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "align_matmul_input_ranks.hpp"
+
+#include "ngraph/op/matmul.hpp"
+#include <ngraph/opsets/opset1.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <transformations/utils/utils.hpp>
+#include <ngraph/pattern/op/or.hpp>
+
+#include <algorithm>
+
+NGRAPH_RTTI_DEFINITION(MKLDNNPlugin::AlignMatmulInputRanks, "AlignMatmulInputRanks", 0);
+
+MKLDNNPlugin::AlignMatmulInputRanks::AlignMatmulInputRanks() {
+    ngraph::OutputVector twoInputs = {
+        ngraph::pattern::any_input(ngraph::pattern::has_static_rank()),
+        ngraph::pattern::any_input(ngraph::pattern::has_static_rank())
+    };
+
+    auto matmulPattern = ngraph::pattern::wrap_type<ngraph::op::MatMul>(twoInputs, ngraph::pattern::has_static_shape());
```

Comment:
Please, remove this check: the transformation supports dynamic shapes and we don't need to check matmul output pshape (we already checked input pshapes)
```suggestion
    auto matmulPattern = ngraph::pattern::wrap_type<ngraph::op::MatMul>(twoInputs);
```

Reply:
Done